### PR TITLE
[REF] hw_posbox_homepage: refactor the homepage to use owl

### PR DESCRIPTION
**Pull request by davidmonnom**
Before this commit, the IOT was using an old system to render pages on the IOT box.

This commit refactors the homepage to use the new owl system.

taskId: 4134163

### Images:
Without advanced settings:
image

With advanced settings:
image

Handler list:
image

Wifi configuration:
image

Server configuration:
image

### CODE REVIEW

**Comment by davidmonnom** [comment]
@qle-odoo Tqt

**Comment by davidmonnom** [comment]
@odoo/rd-security 

Hello, I've done a refactoring of the IOT frontend management to use OWL JS.

Can you add a security override? I haven't changed anything in the backend, just moved a few methods around.

File: addons/hw_posbox_homepage/static/src/app/Homepage.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            console.warn("Error while fetching data");
+        }
+    }
+
+    toggleAdvanced() {
+        this.store.advanced = !this.store.advanced;
+        localStorage.setItem("showAdvanced", this.store.advanced);
+    }
+
+    static template = xml`
```
Comment (lse-odoo) [inline addons/hw_posbox_homepage/static/src/app/Homepage.js:60]: is there not some way to have template XML in dedicated files ? :thinking:

Comment (davidmonnom) [reply addons/hw_posbox_homepage/static/src/app/Homepage.js:60]: If you find it, I will be happy to know

Comment (lse-odoo) [reply addons/hw_posbox_homepage/static/src/app/Homepage.js:60]: I do guess we can't use `web.assets_common` as the IoT does not initiate the JS backend as intended ?

Comment (davidmonnom) [reply addons/hw_posbox_homepage/static/src/app/Homepage.js:60]: Cannot use assets, its a standalone owl application. We don't have any db on the IOT so no assets.

Comment (lse-odoo) [reply addons/hw_posbox_homepage/static/src/app/Homepage.js:60]: correct me if I am wrong, but I thought there was 2 types of assets, the "views" that indeed require a database and the "static" ones which are loaded dynamically. They likely also require a database, but did you double check ?

Comment (davidmonnom) [reply addons/hw_posbox_homepage/static/src/app/Homepage.js:60]: I'm currently thinking about that too, I don't really know, I will try

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: commands to manage the iotbox, assuming it is deployed on a network with no malicious user

@davidmonnom @loouis-t could the security of the deployment be clearly indicated on the documentation of the iot?
I don’t see anything on https://www.odoo.com/documentation/17.0/applications/general/iot.html but maybe it’s somewhere else. We assume an IoT is isolated on a network but it should be clearly mentionned it the scenarios of restaurant/cafés.

**Comment by davidmonnom** [comment]
@mart-e sorry we retarget to 17.0

here's the new PR: 
https://github.com/odoo/odoo/pull/177895

This one will be closed.

**Pull request closed without merging**